language/t5/train_t5.py

```diff
@@ -0,0 +1,118 @@
+from model.t5 import T5LMLoss
+from colossalai.utils import is_using_pp
+
+
+def main():
+    parser = colossalai.get_default_parser()
+    parser.add_argument('--from_torch', default=False, action='store_true')
+    args = parser.parse_args()
+    disable_existing_loggers()
+    if args.from_torch:
+        colossalai.launch_from_torch(config=args.config)
+    else:
+        colossalai.launch_from_slurm(config=args.config,
+                                     host=args.host,
+                                     port=29500,
+                                     seed=42)
+
+    logger = get_dist_logger()
+
+    logger.info('Build data loader', ranks=[0])
+    train_ds = WebtextDataset(os.environ['DATA'], seq_len=gpc.config.SEQ_LEN)
+    train_dataloader = utils.get_dataloader(train_ds,
+                                            seed=42,
+                                            batch_size=gpc.config.BATCH_SIZE,
+                                            pin_memory=True,
+                                            shuffle=True,
+                                            drop_last=True)
+
+    logger.info('Build model', ranks=[0])
+    use_pipeline = is_using_pp()
+    use_interleaved = hasattr(gpc.config.model, 'num_chunks')
+    use_zero3 = hasattr(gpc.config, 'zero') and gpc.config.zero.level == 3
+    ctx = zero3_model_context() if use_zero3 else contextlib.nullcontext()
+    with ctx:
+        model = gpc.config.model.pop('type')(**gpc.config.model)
+    if use_pipeline and use_interleaved and not isinstance(model, nn.ModuleList):
+        model = nn.ModuleList([model])
+
+    criterion = getattr(gpc.config, 'loss_fn', None)
+    if criterion is not None:
+        criterion = criterion.type()
+    else:
+        criterion = T5LMLoss()
+
+    logger.info('Build optimizer', ranks=[0])
+    optimizer = gpc.config.optimizer.pop('type')(
+        model.parameters(), **gpc.config.optimizer)
+
+    lr_scheduler = LinearWarmupLR(
+        optimizer, total_steps=gpc.config.NUM_EPOCHS, warmup_steps=5)
+
+    engine, train_dataloader, _, lr_scheduler = colossalai.initialize(model,
+                                                                      optimizer,
+                                                                      criterion,
+                                                                      train_dataloader=train_dataloader,
+                                                                      lr_scheduler=lr_scheduler)
+    global_batch_size = gpc.config.BATCH_SIZE * \
+        gpc.get_world_size(ParallelMode.DATA) * getattr(gpc.config, "gradient_accumulation", 1)
+    logger.info(f'Init done, global batch size = {global_batch_size}', ranks=[0])
+    tensor_shape = getattr(gpc.config, 'TENSOR_SHAPE', None)
+    schedule = None
+    if use_pipeline:
+        if use_interleaved:
+            logger.info('Build InterleavedPipelineSchedule', ranks=[0])
```

Comment:
Hi @SaraBakic , the latest ColossalAI has refactored `schedule`, can you please make some changes to adapt to it? You can refer to https://github.com/hpcaitech/ColossalAI-Examples/blob/main/language/gpt/train_gpt.py.